cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkProgramRunner.java

```diff
@@ -154,8 +154,8 @@ public ProgramController run(Program program, ProgramOptions options) {
 
       // Setup dataset framework context, if required
       if (programDatasetFramework instanceof ProgramContextAware) {
-        Id.Program programId = program.getId();
-        ((ProgramContextAware) programDatasetFramework).initContext(new Id.Run(programId, runId.getId()));
+        ProgramId programId = program.getId().toEntityId();
```

Comment:
it may be better to not change this in this PR. We're creating 2 extra objects here. Lets change it after ``Program`` and ``ProgramContextAware`` are changed.